File: leetcode/POTD/2025-02-05/solution.go
func areAlmostEqual(s1 string, s2 string) bool {
    r1, r2 := []rune(s1), []rune(s2)
    var diff [][2]rune

    for i := range r1 {
        if r1[i] != r2[i] {
            diff = append(diff, [2]rune{r1[i], r2[i]})
            if len(diff) > 2 {
                return false
            }
        }
    }

    return len(diff) == 0 || (len(diff) == 2 && diff[0][0] == diff[1][1] && diff[0][1] == diff[1][0])
}

File: leetcode/POTD/2025-02-05/solution.rs
impl Solution {
    pub fn are_almost_equal(s1: String, s2: String) -> bool {
        let mut diff = Vec::new();

        for (a, b) in s1.chars().zip(s2.chars()) {
            if a != b {
                diff.push((a, b));
                if diff.len() > 2 {
                    return false;
                }
            }
        }

        diff.is_empty() || (diff.len() == 2 && diff[0] == (diff[1].1, diff[1].0))
    }
}
